demo/apisix/conf/apisix.yaml

```diff
@@ -164,6 +164,14 @@ routes:
       - OPTIONS
     plugin_config_id: jmap-plugin
 
+stream_routes:
+  - id: imap_starttls
+    server_port: 143
+    upstream_id: imap_starttls_upstream
+  - id: imap_ssl
+    server_port: 993
+    upstream_id: imap_ssl_upstream
```

Comment:
Is it possible to do SSL termination on Apisix side?

Reply:
Yes IMO:
https://apisix.apache.org/docs/apisix/stream-proxy/#accept-tls-over-tcp-connection